libminifi/test/unit/FileSystemRepositoryTests.cpp

```diff
@@ -47,7 +50,6 @@ TEST_CASE("Test Physical memory usage", "[testphysicalmemoryusage]") {
     stream->write(fragment.as_span<const std::byte>());
   }
 
-  const auto end_memory = utils::OsUtils::getCurrentProcessPhysicalMemoryUsage();
-
-  REQUIRE(gsl::narrow<size_t>(end_memory - start_memory) < 1_MB);
+  using org::apache::nifi::minifi::utils::verifyEventHappenedInPollTime;
+  REQUIRE(verifyEventHappenedInPollTime(5s, [&] { return gsl::narrow<size_t>(utils::OsUtils::getCurrentProcessPhysicalMemoryUsage() - start_memory) < 5_MB; }, 100ms));
```

Comment:
It's not likely to happen, but I don't think we want to terminate if the current memory usage is lower than `start_memory`.
```suggestion
  CHECK(verifyEventHappenedInPollTime(5s, [&] {
    const auto end_memory = utils::OsUtils::getCurrentProcessPhysicalMemoryUsage();
    REQUIRE(end_memory > 0);
    return end_memory < start_memory + 5_MB;
  }, 100ms));
```

I would also put a `REQUIRE(start_memory > 0)` after line 42.

Reply:
Updated in 45c5e56b0d3804b6fa7f990b4bb3c15666d0de48

Reply:
`gsl::narrow<size_t>(end_memory - start_memory)` still terminates if `end_memory` is less than `start_memory`.  I think `end_memory < start_memory + int64_t{5_MB}` would be better.

Reply:
Oh I see, you are right, fixed in 26e45b1525e76a4ce6d4d2844226868b0c29f068